hopp/simulation/technologies/financial/custom_financial_model.py

```diff
@@ -228,9 +252,155 @@ def execute(self, n=0):
+        sol_nominal = pf_nominal.solve_price()
+        lcoe_nominal = sol_nominal['price']
+
+        usd_per_kwh_to_cents_per_kwh = 100
+        self.value('levelized_cost_of_energy_real', lcoe_real*usd_per_kwh_to_cents_per_kwh)
+        self.value('levelized_cost_of_energy_nominal', lcoe_nominal*usd_per_kwh_to_cents_per_kwh)
+
         return
+    
+    def setup_profast(self, gen_inflation) -> ProFAST.ProFAST:
+
+        """This method sets up a cash-flow financial model based on the input financial parameters.
+
+        Args:
+            gen_inflation (float): inflation is left as an input and should be a float between 0 and 1, not a percentage. 
+            Leaving inflation as an input to the method allows for easily setting up a ProFAST model for either real or nominal calculations.
+
+        Returns:
+            ProFAST.ProFAST: an instance of the ProFAST class set up according to the input financial parameters
+        """
+
+        nominal_discount_rate = self.nominal_discount_rate(
+            inflation_rate=self.value('inflation_rate'),
+            real_discount_rate=self.value('real_discount_rate')
+        ) / 100
+
+        pf = ProFAST.ProFAST()
+        pf.set_params(
+            "commodity",
+            {
+                "name": "Electricity",
+                "unit": "kWh",
+                "initial price": 10,
+                "escalation": gen_inflation,
+            }, 
+        )
+        
+        if "Battery" in self.name:
+            pf.set_params(
+                "capacity",
+                max([1E-6, self.value("batt_annual_discharge_energy")[0]/365.0]),
+            )  # kWh/day
+        else:
+            pf.set_params(
+                "capacity",
+                max([1E-6, self.value("annual_energy_kwh")/365.0]),
+            )  # kWh/day
+
+        pf.set_params("maintenance", {"value": self.o_and_m_cost(), "escalation": gen_inflation})
+
+        # pf.add_fixed_cost(
+        #     name="Fixed O&M Cost",
+        #     usage=1.0,
+        #     unit="$/year",
+        #     cost=self.o_and_m_cost(),
+        #     escalation=gen_inflation,
+        # )
+
+        pf.set_params(
+            "analysis start year", self.value('analysis_start_year'), # no explicit year in single owner,  # Add financial analysis start year
+        )
+        pf.set_params(
+            "operating life", self.value('analysis_period')
+        )
+        pf.set_params(
+            "installation months",
+            self.value('installation_months'),  # Add installation time to yaml default=0
+        )
+        pf.set_params(
+            "installation cost",
+            {
+                "value": 0,
+                "depr type": "Straight line",
+                "depr period": 4,
+                "depreciable": False,
+            },
+        )
+        pf.set_params("demand rampup", 0)
+        pf.set_params("long term utilization", 1)  # TODO should use utilization
```

Comment:
Does it make sense to change this to the capacity factor and use the rated capacity for the `set_param("capacity")`?

Reply:
I'm inclined to keep it as is. Using capacity factor and nameplate capacity is essentially having ProFAST make the same calculations we already did to get the actual capacity. I'm open to discussion here, but let's make it a separate issue/PR.

Reply:
I think it makes sense to just make an issue about it, if we include multi-year simulations we will probably want to be able to add the dictionaries that include performance over time.

Reply:
I have created an issue to continue this discussion: #402